cardinal/ecs/world_logger.go

```diff
@@ -0,0 +1,110 @@
+package ecs
+
+import (
+	"github.com/rs/zerolog"
+	"path/filepath"
+	"pkg.world.dev/world-engine/cardinal/ecs/component"
+	"pkg.world.dev/world-engine/cardinal/ecs/storage"
+	"reflect"
+	"runtime"
+)
+
+type WorldLogger struct {
+	world  *World
+	logger *zerolog.Logger
+}
+
+func NewWorldLogger(logger *zerolog.Logger, world *World) WorldLogger {
+	return WorldLogger{
+		logger: logger,
+		world:  world,
+	}
+}
+
+func (wl *WorldLogger) Log(traceId string, logLevel zerolog.Level, message string) {
+	wl.logger.WithLevel(logLevel).Str("trace_id", traceId)
+}
+
+func (wl *WorldLogger) LogInfo(traceId string, message string) {
```

Comment:
What will traceId be used for?

Did you consider using a context.Context to keep track of trace ids (and/or span ids)?

Reply:
No. Did not kno about this. I will look into it and utilize.

Reply:
I researched go contexts. 

My intention with trace ids are indeed used similarly to a unique id for a go context.

However I think it's better to have a string here. 

1. The reason is these functions can be used in places where there is no context (which in turn requires the user to instantiate the context and set a key, value pair) 

2. context may not have the specific key for the trace_id that the logger is looking for. 

What happens is the user just passes the context to the logger, the logger can't find the trace_id and now the resulting log does not have a trace. The user doesn't have to think about trace ids he just fulfills the function signature. 

By keeping it a string here it forces the user to explicitly choose to either have a trace_id by filling it in with something, or if he doesn't care he can pass a useless or empty string. This choice is abstracted away from the user if we pass a context object. He may not understand even the purpose of the context object without consulting the function definition. 

Better to do this if you want to grab it off the context:

```
Log(ctx.Value('trace_id').....)
```

That's just my opinion. I realize it is a bit wordy to do that on every log What do you think?